### ISSUE TITLE
Build without libgtk/GUI?

### ISSUE THREAD

**Issue by ryanattard** [hgst/libzbc#36]
Is there a way to compile without libgtk? I am hitting this issue referenced earlier: https://github.com/hgst/libzbc/issues/2 but I don't want to build the GUI program and it seems it should be optional based on how the autoconf file is authored.

**Comment by damien-lemoal** [comment]
I added the "--disable-gui" configuration option. Running "configure --disable-gui" should now prevent compilation of the GUI tools.

**Comment by ryanattard** [comment]
Thanks! This helps. The other part I struggled with here was that you need pkg-config for the conditional package flags to work or else you get those syntax errors referenced in #2 . pkg-config is not installed on Debian (tried sid) with build-essential.

**Comment by ryanattard** [comment]
Oh, he didn't have exactly the same trace I had. I had:

`./configure: line 12855: syntax error near unexpected token `GTK,'
./configure: line 12855: `	PKG_CHECK_MODULES(GTK, gtk+-3.0, HAVE_GTK3=1, HAVE_GTK3=0)'`

which goes away if you run autogen.sh with pkg-config installed.

**Comment by damien-lemoal** [comment]
OK. Will see if I can change the use of pkg-config to be optional. I thought that the PKG_CHECK_MODULES() macro would do something intelligent if pkg-config is not installed, but apparently that is not the case (using Fedora and do not have this problem).

**Comment by ryanattard** [comment]
I can post some reproduction commands with docker if you are interested.

**Comment by damien-lemoal** [comment]
Just pushed a commit to avoid configure errors if pkg-config is not installed. Please try it.
It  works for me (tested by manually uninstalling pkg-config).

**Comment by ryanattard** [comment]
Still reproducing it:
```
checking for linux/blkzoned.h... no
./configure: line 12856: syntax error near unexpected token `GTK,'
./configure: line 12856: `      PKG_CHECK_MODULES(GTK, gtk+-3.0, HAVE_GTK3=1, HAVE_GTK3=0)'
```

Resultant relevant configure block (shows your changes):
```
# Conditionals

# Check whether --enable-gui was given.
if test "${enable_gui+set}" = set; then :
  enableval=$enable_gui;
fi

if test "x$enable_gui" != "xno"; then :

        # Build gzbc only if GTK3 is installed.
        PKG_PROG_PKG_CONFIG 
        PKG_CHECK_MODULES(GTK, gtk+-3.0, HAVE_GTK3=1, HAVE_GTK3=0)
         if test "$HAVE_GTK3" -eq 1; then
  BUILD_GUI_TRUE=
  BUILD_GUI_FALSE='#'
else
  BUILD_GUI_TRUE='#'
  BUILD_GUI_FALSE=
fi
  
         if test "$HAVE_GTK3" -eq 1; then
  BUILD_GUI_TRUE=
  BUILD_GUI_FALSE='#'
else 
  BUILD_GUI_TRUE='#'
  BUILD_GUI_FALSE=
fi
```
There's this thread https://github.com/curl/curl/issues/972 I was looking at to author a patch but I am really not that familiar with autotools and I couldn't quite get it to work. 

I can reproduce it with this docker command: 
```
docker run -it -v ~/libzbc:/libzbc --workdir /libzbc debian:buster bash -c 'apt-get update && apt-get -y install build-essential autoconf automake libtool && sh autogen.sh && ./configure && make'
```

**Comment by damien-lemoal** [comment]
Yes, it is worth a  try...
In configure.ac, replace this:

AS_IF([test "x$enable_gui" != "xno"],
[
        # Build gzbc only if GTK3 is installed.
        PKG_PROG_PKG_CONFIG
        PKG_CHECK_MODULES([GTK], [gtk+-3.0], [HAVE_GTK3=1], [HAVE_GTK3=0])
        AM_CONDITIONAL([BUILD_GUI], [test "$HAVE_GTK3" -eq 1])
],
[
        AM_CONDITIONAL([BUILD_GUI], false)
])

with:

AS_IF([test "x$enable_gui" != "xno"],
[
        # Build gzbc only if GTK3 is installed.
        m4_ifdef([PKG_CHECK_MODULES],
                       [PKG_CHECK_MODULES([GTK], [gtk+-3.0], [HAVE_GTK3=1], [HAVE_GTK3=0])],
                       [HAVE_GTK3=0])
        AM_CONDITIONAL([BUILD_GUI], [test "$HAVE_GTK3" -eq 1])
],
[
        AM_CONDITIONAL([BUILD_GUI], false)
])

**Comment by ryanattard** [comment]
That worked, the part that seemed of most consequence was that it links a hard dependency between pkg-config and gtk3 components. Is that an acceptable tradeoff?

**Comment by damien-lemoal** [comment]
From what I read about PKG_CHECK_MODULES, the errors you were experiencing are due to pkg-config not being installed in the "default" location and PKG_CHECK_MODULES fails as it calls it with an incorrect path. The test "m4_ifdef" apparently avoid this error.

I pushed this fix for now since it works for you and I did not find any problem on my system. Let's see how it goes and I will revisit this if others report problems.
